From 8d1f23f5135ae0a4f191af898000abe02aa64910 Mon Sep 17 00:00:00 2001
From: Thomas Parrott <thomas.parrott@canonical.com>
Date: Sat, 11 Dec 2021 22:35:03 +0000
Subject: [PATCH] lxd/device/nic/routed: Enable IP forwarding on veth host_name
 interface

For IPv4 traffic this means we don't need to require net.ipv4.conf.all.forwarding=1.

Fixes #9678

Signed-off-by: Thomas Parrott <thomas.parrott@canonical.com>
---
 lxd/device/nic_routed.go | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/lxd/device/nic_routed.go b/lxd/device/nic_routed.go
index ebbebedb2..1b82ec1a7 100644
--- a/lxd/device/nic_routed.go
+++ b/lxd/device/nic_routed.go
@@ -306,6 +306,12 @@ func (d *nicRouted) Start() (*deviceConfig.RunConfig, error) {
 			if err != nil {
 				return nil, fmt.Errorf("Failed adding host gateway IP %q: %w", addr.Address, err)
 			}
+
+			// Enable IP forwarding on host_name.
+			err = util.SysctlSet(fmt.Sprintf("net/%s/conf/%s/forwarding", keyPrefix, saveData["host_name"]), "1")
+			if err != nil {
+				return nil, err
+			}
 		}
 
 		// Perform per-address host-side configuration (static routes and neighbour proxy entries).
-- 
2.34.1

